anvill/src/TypeParser.cpp

```diff
@@ -541,6 +543,11 @@ llvm::Expected<llvm::Type *> ParseType(llvm::LLVMContext &context,
     }
 
     auto type = remill::GetReference(ret);
+    if (type->isFunctionTy()) {
```

Comment:
I think the better approach is to always return the type, regardless of if it is a function/sized/unsized type, and instead do the `isSized` check at the usage site of `ParseType`. Right now, that logic is basically having to happen with the `is_function` checks at the usage site, so really, that tells me that the `isSized()` check doesn't belong here anymore.